ocs_ci/utility/multicluster.py

```diff
@@ -237,11 +202,52 @@ def generate_pytest_parameters(self, metafunc, roles):
         """
         pytest_params = []
         for role in roles:
-            pytest_params.extend(self.get_pytest_params_tuple(role))
+            # A role marker may have mdr, rdr etc markers, we need to pick
+            # the role marker only if its applicable to the current DR scenario
+            if role.startswith(f"{self.dr_type}"):
+                pytest_params.extend(self.get_pytest_params_tuple(role))
         return pytest_params
 
 
-multicluster_upgrade_parametrizer = {"metro-dr": MDRClusterUpgradeParametrize}
+class MDRClusterUpgradeParametrize(MultiClusterUpgradeParametrize):
+    """
+    This child class handles MDR upgrade scenario specific pytest parametrization
+
+    """
+
+    def __init__(self):
+        self.dr_type = "mdr"
+        super().__init__()
+        self.all_roles = MDR_ROLES
+
+    def config_init(self):
+        super().config_init()
+
+
+class RDRClusterUpgradeParametrize(MultiClusterUpgradeParametrize):
+    """
+    This child class handles RDR upgrade scenario specific pytest parametrization
+
+    """
+
+    def __init__(self):
+        self.dr_type = "rdr"
+        super().__init__()
+        self.all_roles = RDR_ROLES
+        # If the current run includes PassiveACM then we need to add
+        # it to the list as by default RDR Roles list will not have PassiveACM
+        for cluster in get_all_acm_indexes():
+            if cluster != get_active_acm_index():
+                self.all_roles.append("PassiveACM")
```

Comment:
Not sure, if there might be some issue with that, but what about using `get_passive_acm_index` function?
```
if get_passive_acm_index():
    self.all_roles.append("PassiveACM")
```

Reply:
done